Lib/test/test_tkinter/test_widgets.py

```diff
@@ -1377,6 +1377,11 @@ class MenuTest(AbstractWidgetTest, unittest.TestCase):
     def create(self, **kwargs):
         return tkinter.Menu(self.root, **kwargs)
 
+    def test_indexcommand_none(self):
+        widget = self.create()
+        i = widget.index('none')
+        self.assertIsNone(i)
```

Comment:
There should be a test that fails without the patch, in addition to testing the current behavior.
```suggestion
        i = widget.index('none')
        self.assertIsNone(i)
        i = widget.index('')
        self.assertIsNone(i)
```

Reply:
The test as I had written it **does** fail without the patch, but only for Tk 8.7 (which the CI obviously is not using). The test is not expected to fail under Tk 8.6.

I had created this test to pass `"none"` to the `index` command because that is the reserved way of specifying the none index: https://www.tcl-lang.org/man/tcl8.7/TkCmd/menu.html#M27. Empty string is not reserved, and is actually a valid label for a menu entry and so a valid argument for the `index` command; if there is no menu entry labeled `''`', then there will instead be a `bad menu entry index ""` error from Tk. So I do not see why the second test should be added.

Reply:
>  Empty string is not reserved, and is actually a valid label for a menu entry and so a valid argument for the `index` command; if there is no menu entry labeled `""`, then there will instead be a `bad menu entry index ""` error from Tk.

To correct myself: Tk 8.7 now also reserves empty string for specifying the none index (in addition to `"none"`). An entry can still be labeled `""`, but such an entry is arguably not very useful either.

Reply:
The new test failed as you expected and I removed it.  I see now that there is a difference between what one might reasonably pass to index() and what the call might return.  You patch is only about preparing for '' as a possible return value.